Added 'Getting started' section (#402)
crystal-lang/crystal-book: getting_started/README.md

# Getting started

Hi and welcome to Crystal's Reference Book!

> First we are going to check if we have [installed the language](https://crystal-lang.org/install/) correctly so that we may try all the examples listed in this book.

Reviewer: The text shouldn't assume readers have already installed the language, might have come here first. Let's guide them to the installation instructions.

```suggestion
First let's make sure to [install the compiler](https://crystal-lang.org/install/) correctly so that we may try all the examples listed in this book.
```

Once installed, the Crystal compiler should be available as `crystal` command.

Let's try it!

## Crystal version

We may check the Crystal compiler version. If Crystal is installed correctly then we should see something like this:

```terminal-session
$ crystal --version

> Crystal 0.34.0 (2020-04-07)

Reviewer: @bcardiff Can we establish a step in the dist process to make sure this is updated on a new release to always show the latest version? I'm not sure if that should just be an item on the manual TODO list or could be an automatic step.

Reviewer 2: Sure, but once we move out from gitbook. To avoid some solution that is might not be the best fit on a future stack.

Reviewer: Hm, updating a markdown file should be workable in any case. Or do you have some client-side expansion with current data in mind? That seems like overkill.

Reviewer 2: Currently there is no publish of the book after the release.

For a couple of releases I'm fine manually updating this.

At some point it is another case of how to validate the code samples are still valid.

Having a script that will update the .md in the repo will imply some magic string or ad-hoc process. Between doing that for one case and manually editing it, I prefer the latter.

LLVM: 10.0.0
Default target: x86_64-apple-macosx
```

Great!

## Crystal help

> Now, if we want to list all the options given by the compiler, we may run `crystal` and that's it:

Reviewer: ```suggestion
Now, if we want to list all the options given by the compiler, we may run `crystal` program without any arguments:
```

```terminal-session
$ crystal
Usage: crystal [command] [switches] [program file] [--] [arguments]

Command:
    init                     generate a new project
    build                    build an executable
    docs                     generate documentation
    env                      print Crystal environment information
    eval                     eval code from args or standard input
    play                     starts Crystal playground server
    run (default)            build and run program
    spec                     build and run specs (in spec directory)
    tool                     run a tool
    help, --help, -h         show this help
    version, --version, -v   show version

Run a command followed by --help to see command specific information, ex:
    crystal <command> --help
```

> We have a section with some examples about [using the compiler](../using_the_compiler/README.md).

Reviewer: The manpage should also be mentioned.
```suggestion
More details about using the compiler can be found on the manpage `man crystal` or in our [compiler manual](../using_the_compiler/README.md).
```

# Getting started

Hi and welcome to Crystal's Reference Book!

First we are going to check if we have [installed the language](https://crystal-lang.org/install/) correctly so that we may try all the examples listed in this book.

Once installed, the Crystal compiler should be available as `crystal` command.

Let's try it!

## Crystal version

We may check the Crystal compiler version. If Crystal is installed correctly then we should see something like this:

```terminal-session
$ crystal --version

> Crystal 0.33.0 (2020-02-14)

Reviewer: This is already outdated at print time 😄

Author: 🙈

> Crystal 0.33.0 (2020-02-14)

Reviewer 2: ```suggestion
Crystal 0.34.0 (2020-04-07)
```

> LLVM: 9.0.1

Reviewer 2: ```suggestion
LLVM: 10.0.0
```
